### PR TITLE
Adding search and user profile

### PR DESCRIPTION
**Pull request by miguelvazq**
WIP React frame + navigation + search + user profile manager.

Signed-off-by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: esp/src/src/react/components/ProfileManager.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        }
+    }
+
+    React.useEffect(() => {
+        if (prevOpen.current === true && open === false) {
+            anchorRef.current!.focus();
+        }
+
+        prevOpen.current = open;
+    }, [open]);
```
Comment (GordonSmith) [inline esp/src/src/react/components/ProfileManager.tsx:102]: See the "usePrevious" pattern here:  https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

File: esp/src/src/react/components/ProfileManager.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                                        <Divider />
+                                        <MenuItem onClick={handleClose}><Typography color="error">Logout</Typography></MenuItem>
+                                    </MenuList>
+                                </div>
+                            </ClickAwayListener>
+                        </Paper>
+                    </Grow>
+                )}
+            </Popper>
+        </>
```
Comment (GordonSmith) [inline esp/src/src/react/components/ProfileManager.tsx:157]: Tranlations...

File: esp/src/src/react/components/ProfileManager.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            minWidth: "200px",
+            marginTop: theme.spacing(2)
+        },
+        userDetails: {
+            padding: theme.spacing(3, 0, 1, 0)
+        }
+  })
+);
+
+export const ProfileManager: React.FC<ProfileManagerProps> = (props) => {
```
[the remark this answers is not in the document]
Comment (miguelvazq) [reply esp/src/src/react/components/ProfileManager.tsx:72]: Yes.

File: esp/src/src/react/components/GlobalSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        }
+    },[searchTerm]);
+
+    React.useEffect(() => {
+        if (prevOpen.current === true && open === false) {
+            anchorRef.current!.focus();
+        }
+
+        prevOpen.current = open;
+    }, [open]);
```
[the remark this answers is not in the document]
Comment (miguelvazq) [reply esp/src/src/react/components/GlobalSearch.tsx:112]: Are you asking about the prevOpen or anchorRef? For open/not states I have created a simple hook in our hooks folder for remembering the previous value which uses the useRef hook.

File: esp/src/src/react/components/ProfileManager.tsx (hunk 1 above)
Comment (miguelvazq) [reply esp/src/src/react/components/ProfileManager.tsx]: Added new hook in hooks folder to cover this.

**Comment by miguelvazq** [comment]
@GordonSmith updated

**Pull request merged**